information/TASK.md

```diff
@@ -57,6 +52,99 @@ export function model (): tf.LayersModel {
   return model
 ```
 
+#### Preprocessing
+
+In the Task object we can optionally choose to add preprocessing functions. Preprocessing is defined [here](../discojs/src/dataset/preprocess.ts),
+and is currently only implemented for images. The predefined functions (resize, normalize, ...) can be found there, and it is also there
+where you can easily create custom preprocessing.
+
+> Preprocessing is done every time an image is read, this will increase the overhead of training since image loading is lazy; for better performance use data that is already preprocessed.
+
+Suppose we want our custom preprocessing that divides our pixels by 2. In the [preprocessing](../discojs/src/dataset/preprocess.ts) file, 
+first we add the enum of our custom function:
+
+```js
+export enum Preprocess {
+  Normalize = 'normalize',
+  Resize = 'resize',
+  Custom = 'custom'
+}
+```
+
+Then we define our custom function
+
+```js
+function custom (image: tf.Tensor3D): tf.Tensor3D {
+return image.div(tf.scalar(2))
+}
+```
+
+And then we include said function to the preprocessing
+
+```js
+export function getPreprocessImage (info: TrainingInformation): PreprocessImage {
+  const preprocessImage: PreprocessImage = (image: tf.Tensor3D): tf.Tensor3D => {
+    ...
+    ...
+    ...
+    if (info.preprocessFunctions.includes(Preprocess.Custom)) {
+      image = custom(image)
+    }
+    return image
+  }
+  return preprocessImage
+}
+```
+
+Finally, in our task we need to add our custom preprocessing
+
+```js
+import { Preprocess } from '../dataset/preprocess'
+
+export const task: Task = {
+  taskID: 'My_task',
+  ...
+  ...
+    trainingInformation: {
+    ...
+    ...
+    preprocessFunctions: [Preprocess.Custom],
+    ...
+    ...
+  }
+}
+```
+
+### Model
+
+The model is built at the end of the file and must be defined as a function called `model` that returns a `tf.LayersModel`.
+
+If we want to define a model programmatically we can do as follows:
+
+```js
+export function model (): tf.LayersModel {
+  // Init model
+  const model = tf.sequential()
+
+  // Add layers
+  model.add(...)
+
+  return model
+
+```
+
+Alternatively we can also load a pre-existing model:
+
+```js
+export async function model (): Promise<tf.LayersModel> {
+  return await tf.loadLayersModel('file:./modelPath/model.json')
+}
+```
+
+the `modelPath` is relative to the root directory of the `server`.
+
+### Making the task visible to the API
+
 After adding the `simple_face.ts` task class, we also need to export it in the `index.ts` file which lives in the same folder.
```

Comment:
maybe we can rename it to `my_new_task.ts` or sth similar, as `simple_face` already exists so might be confusing? or do you mean to edit an existing task or duplicate to a fresh one?